KAS-3606 Use new authentication model

erikap opened the pull request:
Update backend to use new authentication model. Required information to determine access rights for a user are now attached to the session during login. 

Changes include:
- bump mock-login-service
- bump acmidm-login-service
- bump cache-warmup-service
- configure new user data model in mu-cl-resources
- update authorization config:
  - add mapping of roles to groups
  - update query to fetch a user's roles based on the information attached to the session 
  - reshuffle group specs in a more logical order in `config.ex`
- update eager indexing groups in search config due to updated authorization groups 
- migration with new mock accounts
- add proxy service in development mode to facilitate testing of ACM/IDM login on local environment

Services that require a release before merging this PR:
- [x] mock-login-service (https://github.com/kanselarij-vlaanderen/mock-login-service/pull/1)
- [x] acmidm-login-service (https://github.com/kanselarij-vlaanderen/acmidm-login-service/pull/10)
- [x] cache-warmup-service (https://github.com/kanselarij-vlaanderen/cache-warmup-service/pull/8)

**Note: this PR may only be merged/deployed after ACM/IDM has upgraded the user roles on their side. Otherwise, all users will login as being 'Overheidsorganisatie' when using ACM/IDM.**

erikap committed:
Add migration to remove old mock accounts

erikap committed:
Add mock-accounts for new user roles

erikap committed:
Remove old account groups

erikap committed:
Configure updated user data model in resource, dispatcher and authorization

erikap committed:
Bump mock-login service

erikap committed:
Update query to validate access by role in mu-authorization config

erikap committed:
Merge branch 'feature/KAS-3602-user-roles-codelist' into feature/KAS-3605-mock-accounts

erikap committed:
Fix query to validate access rights in mu-authorization

erikap committed:
Update base URIs of mock accounts

ValenberghsSven committed:
bump frontend

erikap committed:
Add proxy to test app locally over HTTPS with domain name in dev mode

erikap committed:
Configure user data to be available to authenticated users

Reshuffle authorization config and use consistent naming for group
specs

ValenberghsSven committed:
Merge branch 'acceptance' of github.com:kanselarij-vlaanderen/app-kaleidos into development

erikap committed:
Bump ACM/IDM login service

erikap committed:
Fix typo in mu-cl-resource config

erikap committed:
Update testdata ZIP files

sergiofenoll committed:
fix: use concept-scheme for government fields URI

For all other Themis URIs we use concept-scheme, not concept-schema. The
URIs are replaced for consistency.

sergiofenoll committed:
feat: add government domain triples

Includes all government domains from 2006 til the present. The
prefLabels contain the provided data, not what is already present in the
database (e.g. "Financiën en begroting" instead of "Financiën en
Begroting").

sergiofenoll committed:
feat: migrate existing government domains to Themis version

sergiofenoll committed:
feat: add government field triples

sergiofenoll committed:
feat: migrate existing government fields to Themis version

tdn committed:
updated migrations after Erika's remarks

ValenberghsSven committed:
bump frontend for hotfix searchdates clearing

ValenberghsSven committed:
Merge branch 'production' of github.com:kanselarij-vlaanderen/app-kaleidos into acceptance

ValenberghsSven committed:
Merge branch 'acceptance' of github.com:kanselarij-vlaanderen/app-kaleidos into development

sergiofenoll committed:
feat: add {start,end}-date properties to concept resource

sergiofenoll committed:
fix: overwrite rdf:type of government areas

sergiofenoll committed:
fix: use old names for government domains

This way we still have the old capitalization.

erikap committed:
Merge pull request #313 from kanselarij-vlaanderen/feature/KAS-3602-user-roles-codelist

KAS-3602 user roles codelist

tdn reviewed (approved):
All looks good as far as I can tell. I suppose we'll only be able to test this thoroughly after ACM-IDM has implemented the changes and we deploy this on accept...

I like the new model structure with the memberships. Will require some revising of the follow-up stories, but it is a lot cleaner.

sergiofenoll committed:
feat: migrate pubflow provisoir government areas

?pubflow pub:beleidsdomein#provisioir ?governmentArea triples are
removed and in their place the government area has been moved to the
respective decisionmaking-flow and subcase.

erikap committed:
Bump cache-warmup-service

erikap committed:
Fix authorization config: eli:LegalResource must be writable in kanselarij-graph

sergiofenoll committed:
fix: add missing comma to authorization config

erikap committed:
Merge pull request #318 from kanselarij-vlaanderen/fix/auth-config-comma

Fix missing comma

erikap committed:
Fix authorization config: ext:SysteemNotificatie must be writable for admins

erikap committed:
Enable error on unwritten data in mu-authorization

erikap committed:
Bump frontend

erikap committed:
Merge branch 'acceptance' into development

erikap committed:
Cleanup ext:Beleidsveld and ext:Beleidsdomein RDF types

erikap committed:
Merge branch 'development' into feature/KAS-3520-government-fields-register

erikap committed:
Bump publication-report-service

erikap committed:
Merge pull request #316 from kanselarij-vlaanderen/feature/KAS-3520-government-fields-register

KAS-3520: government fields register

erikap committed:
Merge pull request #317 from kanselarij-vlaanderen/feature/KAS-3576-migrate-pubflow-provisoir-government-areas

KAS-3576: migrate provisional government areas

erikap committed:
Merge branch 'development' of github.com:kanselarij-vlaanderen/app-kaleidos into development

MikiDi committed:
Bump cache-warmup

MikiDi committed:
Merge branch 'acceptance' into development

erikap committed:
Merge branch 'development' into feature/KAS-3606-user-roles

erikap committed:
Fix typos in user role notations

erikap committed:
Fix inline persagenda documentation in resource config

erikap committed:
Add clarification to meeting-agenda relations in resource config

erikap committed:
Remove unused property from agenda resource config

erikap committed:
Merge branch 'development' into feature/KAS-3606-user-roles

erikap committed:
Use released versions of login, mock-login and cache-warmup
